mindsdb/integrations/handlers/chromadb_handler/chromadb_handler.py

```diff
@@ -316,44 +314,86 @@ def select(
         """
+        collection = self._client.get_or_create_collection(collection_name)
 
-        collection = self._client.get_or_create_collection(name=table_name)
-
-        # drop columns with all None values
+        # Convert metadata from string to dict if needed
+        if TableField.METADATA.value in df.columns:
+            df[TableField.METADATA.value] = df[TableField.METADATA.value].apply(
+                self._dataframe_metadata_to_chroma_metadata
+            )
+            # Drop rows where metadata conversion failed
+            df = df.dropna(subset=[TableField.METADATA.value])
 
-        data.dropna(axis=1, inplace=True)
+        # Convert embeddings from string to list only if they are strings
+        if TableField.EMBEDDINGS.value in df.columns and df[TableField.EMBEDDINGS.value].dtype == 'object':
+            df[TableField.EMBEDDINGS.value] = df[TableField.EMBEDDINGS.value].apply(
+                lambda x: ast.literal_eval(x) if isinstance(x, str) else x
+            )
 
-        def dataframe_metadata_to_chroma_metadata(metadata: Union[Dict[str, str], str]) -> Optional[Dict[str, str]]:
-            if isinstance(metadata, dict):
-                if not metadata:
-                    # ChromaDB does not support empty metadata dicts, but it does support None.
-                    # Related: https://github.com/chroma-core/chroma/issues/791.
-                    return None
-                return metadata
-            # Metadata is a string representation of a dictionary instead.
-            return ast.literal_eval(metadata)
+        # Validate embedding dimensions
+        if TableField.EMBEDDINGS.value in df.columns:
+            embedding_dims = [len(emb) for emb in df[TableField.EMBEDDINGS.value]]
+            if len(set(embedding_dims)) > 1:
+                raise Exception("All embeddings must have the same dimension")
 
-        # ensure metadata is a dict, convert to dict if it is a string
-        if data.get(TableField.METADATA.value) is not None:
-            data[TableField.METADATA.value] = data[TableField.METADATA.value].apply(dataframe_metadata_to_chroma_metadata)
+            # Get existing collection dimension if any records exist
+            if collection.count() > 0:
+                existing = collection.get(include=['embeddings'])
+                if existing and existing['embeddings']:
+                    existing_dim = len(existing['embeddings'][0])
+                    if existing_dim != embedding_dims[0]:
+                        raise Exception(f"Embedding dimension {embedding_dims[0]} does not match collection dimensionality {existing_dim}")
 
-        # convert to dict
+        # For SELECT statements without IDs, deduplicate based on content
+        if TableField.ID.value not in df.columns:
+            df = df.drop_duplicates(subset=[TableField.CONTENT.value])
+            df[TableField.ID.value] = [str(uuid.uuid4()) for _ in range(len(df))]
```

Comment:
Just in case: if `insert` is called from KB controller - it always has ID column